vault/cors.go

```diff
@@ -0,0 +1,137 @@
+package vault
+
+import (
+	"errors"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+var errCORSNotConfigured = errors.New("CORS is not configured")
+
+type CORSConfig struct {
+	enabled          bool
+	allowedOrigins   *[]string
+	allowedHeaders   *map[string]string
```

Comment:
Those are a great example of "predictive engineering". As I was coding I saw a use case to have methods to add new headers, but then realized that was over-engineering things. They should (will) be removed.